Core/WebViewController.swift

```diff
@@ -295,55 +302,56 @@ extension WebViewController: WKNavigationDelegate {
         let error = error as NSError
         if let url = loadedURL,
             let domain = url.host,
-            error.code == webViewErrorCodes.frameLoadInterrupted {
+            error.code == Constants.frameLoadInterruptedErrorCode {
             failingUrls.insert(domain)
         }
         showErrorLater()
     }
     
     public func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        let decision = decidePolicyFor(navigationAction: navigationAction)
+        if decision == .allow && navigationAction.isTargettingMainFrame() {
+            showProgressIndicator()
+        }
+        decisionHandler(decision)
+    }
+    
+    private func decidePolicyFor(navigationAction: WKNavigationAction) -> WKNavigationActionPolicy {
```

Comment:
This logic has grown over time, can we extract this method and unit test it? Happy to capture it in another PR if that makes more sense.
Optional: Does it make sense to return a boolean here?

Reply:
Since there's already constants I think they are better than booleans anyway - boolean return types can often seem ambiguous, imho.

I had thought of writing tests for this but I think it's a rabbit hole in terms of dependencies. I really want to refactor this whole class in to a bunch unit testable components as part of the work to consolidate the TabViewController/WebViewController code so would rather do it as part of that task: https://app.asana.com/0/414709148257752/542099048519218